core/templates/pages/exploration-editor-page/services/user-exploration-permissions.service.spec.ts

```diff
@@ -16,64 +16,66 @@
 
-describe('User Exploration Permissions Service', function() {
-  var ueps, ContextService, UrlService, $httpBackend;
   var sampleExplorationId = 'sample-exploration';
   var samplePermissionsData = {
     canEdit: false,
     canVoiceOver: true,
   };
-  beforeEach(angular.mock.module('oppia'));
-  beforeEach(angular.mock.module('oppia', function($provide) {
-    var ugs = new UpgradedServices();
-    for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
-      $provide.value(key, value);
-    }
-  }));
 
-  beforeEach(angular.mock.inject(function($injector) {
-    ueps = $injector.get('UserExplorationPermissionsService');
-    $httpBackend = $injector.get('$httpBackend');
+  beforeEach(angular.mock.inject(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule]
+    });
 
-    ContextService = $injector.get('ContextService');
-    UrlService = $injector.get('UrlService');
-    spyOn(ContextService, 'getExplorationId').and.returnValue(
+    httpTestingController = TestBed.get(HttpTestingController);
+    ueps = TestBed.get(UserExplorationPermissionsService);
+    contextService = TestBed.get(ContextService);
+    spyOn(contextService, 'getExplorationId').and.returnValue(
       sampleExplorationId);
+    UserExplorationPermissionsService.permissionsPromise = null;
   }));
 
-  it('should fetch the correct data', function() {
-    $httpBackend.expect(
-      'GET', '/createhandler/permissions/' + sampleExplorationId).respond(
-      200, samplePermissionsData);
+  afterEach(()=> {
+    httpTestingController.verify();
+  });
 
+  it('should fetch the correct data', fakeAsync(() => {
     ueps.getPermissionsAsync().then(function(response) {
       expect(response).toEqual(samplePermissionsData);
     });
-  });
 
-  it('should cache rights data', function() {
-    $httpBackend.expect(
-      'GET', '/createhandler/permissions/' + sampleExplorationId).respond(
-      200, samplePermissionsData);
-    ueps.getPermissionsAsync();
-    $httpBackend.flush();
+    let req = httpTestingController.expectOne(
+      '/createhandler/permissions/' + sampleExplorationId);
+    expect(req.request.method).toEqual('GET');
+    req.flush(samplePermissionsData);
+    flushMicrotasks();
+  }));
 
-    $httpBackend.when(
-      'GET', '/createhandler/permissions/' + sampleExplorationId).respond(
-      200, {canEdit: true, canVoiceOver: false});
+  it('should cache rights data', fakeAsync(() => {
     ueps.getPermissionsAsync();
+    let req = httpTestingController.expectOne(
+      '/createhandler/permissions/' + sampleExplorationId);
+    expect(req.request.method).toEqual('GET');
+    req.flush(samplePermissionsData);
+    flushMicrotasks();
 
-    expect($httpBackend.flush).toThrowError('No pending request to flush !');
-  });
+    ueps.getPermissionsAsync();
+    httpTestingController.expectNone(
+      '/createhandler/permissions/' + sampleExplorationId);
```

Comment:
What does this line means?

Reply:
This is to check that there is no http request at the url `/createhandler/permissions/<sampleExplorationId>`.

Reply:
Thanks for the explanation!